bpf/lib/common.h

```diff
@@ -1106,6 +1123,35 @@ struct lpm_val {
 	__u8 flags;
 };
 
+#if defined(ENABLE_DSR) && DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+struct geneve_opt4 {
+	__be16	opt_class;
+	__u8	type;
+	__u8	length:5;
+	__u8	r3:1;
+	__u8	r2:1;
+	__u8	r1:1;
+	__be32	addr;
+	__be16	port;
+	__u16	pad;
+};
+
+struct geneve_opt6 {
+	__be16	opt_class;
+	__u8	type;
+	__u8	length:5;
+	__u8	r3:1;
+	__u8	r2:1;
+	__u8	r1:1;
+	__be32	addr1;
+	__be32	addr2;
+	__be32	addr3;
+	__be32	addr4;
```

Comment:
Have you tried turning this into a `union v6addr`, same as for the IPv6 Extension header? Then we can also use `ipv6_addr_copy()`.